aptos-move/aptos-vm-types/src/change_set.rs

```diff
@@ -60,71 +175,167 @@ impl VMChangeSet {
         Ok(change_set)
     }
 
-    pub fn write_set(&self) -> &WriteSet {
-        &self.write_set
+    /// Returns a new change set, built from the storage representation. Note
+    /// that this is a rather expensive operation because it re-splits resources
+    /// from modules and is only used to support transactions with write-set
+    /// payload.
+    pub fn try_from_storage_change_set(
+        change_set: StorageChangeSet,
+        checker: &dyn CheckChangeSet,
+    ) -> anyhow::Result<Self, VMStatus> {
+        let (write_set, events) = change_set.into_inner();
+
+        // There should be no aggregator writes if we have a change set from
+        // storage.
+        let mut resource_write_set = StateChange::empty();
+        let mut module_write_set = StateChange::empty();
+
+        // Go over write ops to split them into resources and modules.
+        for (state_key, write_op) in write_set {
+            if matches!(state_key.inner(), StateKeyInner::AccessPath(ap) if ap.is_code()) {
+                module_write_set.insert(state_key, write_op);
+            } else {
+                // Everything else must be a resource.
+                // TODO(aggregator): Aggregator V1 is a table item, and so fits into
+                // this category. In practice this should never happen, but we might
+                // want to have an assert here before Aggregator V2 lands.
```

Comment:
hmm, what does go into aggregator_write_set, if not deltas and not writes to v1 aggregator table item?

Reply:
Nothing? We construct from storage here, there are no aggregators there